src/opm/input/eclipse/Schedule/RFTConfig.cpp

```diff
@@ -21,196 +21,281 @@
 
 #include <algorithm>
 #include <cassert>
+#include <optional>
 #include <stdexcept>
+#include <string>
 #include <utility>
 
+namespace {
+
+template <typename Map, typename Predicate>
+bool mapContains(const Map& map, Predicate&& p)
+{
+    return std::any_of(map.begin(), map.end(), std::forward<Predicate>(p));
+}
+
+template <typename Map, typename Predicate>
+void pruneFromMapIf(Map& map, Predicate prune)
+{
+    for (auto begin = map.begin(); begin != map.end(); ) {
+        if (! prune(*begin)) {
+            ++begin;
+            continue;
+        }
+
+        begin = map.erase(begin);
+    }
```

Comment:
> big fan of [reversing the clauses]

Fair enough.  I wrote it this way originally because the common case here is for `prune` to not fire.  I've pushed an update that uses your preferred alternative.